frame/support/src/storage/child.rs

```diff
@@ -19,14 +19,29 @@
 //! This module is a currently only a variant of unhashed with additional `storage_key`.
 //! Note that `storage_key` must be unique and strong (strong in the sense of being long enough to
 //! avoid collision from a resistant hash function (which unique implies)).
+//!
+//! A **key collision free** unique id is required as parameter to avoid key collision
+//! between child tries.
+//! This unique id management and generation responsability is delegated to pallet module.
 // NOTE: could replace unhashed by having only one kind of storage (root being null storage key (storage_key can become Option<&[u8]>).
 
 use crate::rstd::prelude::*;
 use codec::{Codec, Encode, Decode};
 
+/// Unique id managed by the system is not implemented yet.
+const SOME: &'static str = "Managed child unique ids are not yet implemented";
```

Comment:
I was thinking removing this unimplemented part as well. I think all the `Option<UniqueID>` should be changed for UniqueID directly.
Later if we reorganize differently by having this unique ID generated by system or any other way, we can make it optional.